store/index.ts

```diff
@@ -69,3 +69,24 @@ export const mutations: MutationTree<RootState> = {
     Vue.set(state.socket, 'reconnectError', true);
   }
 };
+
+export const actions: ActionTree<RootState, RootState> = {
+  async reset({ dispatch }, { clearCritical }: { clearCritical: boolean }) {
+    const promises = [];
+    promises.push(dispatch('backdrop/clearBackdrop', { root: true }));
+    promises.push(dispatch('deviceProfile/clearDeviceProfile', { root: true }));
+    promises.push(dispatch('displayPreferences/resetState', { root: true }));
+    promises.push(dispatch('homeSection/clearHomeSection', { root: true }));
+    promises.push(dispatch('page/clearPage', { root: true }));
+    promises.push(dispatch('playbackManager/stop', { root: true }));
+    if (clearCritical) {
+      promises.push(dispatch('servers/clearServers', { root: true }));
+    }
+    promises.push(dispatch('snackbar/resetMessage', { root: true }));
+    promises.push(dispatch('tvShows/clearTvShows', { root: true }));
+    promises.push(dispatch('user/clearUser', { root: true }));
+    promises.push(dispatch('userViews/clearUserViews', { root: true }));
+
+    await Promise.all(promises);
```

Comment:
Sorry for not pointing it out before, but why ``Promise.all``? ``await`` at each of the dispatches should also run each of them after the last one, without the need of pushes and arrays.

Reply:
It's not doing the same thing. Doing `await` before every dispatch means that we wait for the first one to finish before starting the second. `Promise.all` returns a promise that's resolved when all of the promises passed as argument are resolved. So here we launch the different dispatch concurrently and wait for them all to finish. The speed gain here is rather small, but there's no need to wait for each and every one of them before starting the next one ;)